webapp/channels/src/packages/mattermost-redux/src/store/configureStore.ts

```diff
@@ -46,7 +46,13 @@ export default function configureStore<S extends GlobalState>({
         autoPause: true,
     });
 
-    const middleware = applyMiddleware(thunk);
+    const middleware = applyMiddleware(
+
+        // @hmhealey I've added this extra argument to Thunks to store information related to the store that can't be
+        // part of Redux state itself. At the moment, this is so that I can attach let DataLoaders dispatch actions.
+        // If you want to make use of this, talk to me first since I want to know more.
+        thunk.withExtraArgument({loaders: {}}),
```

Comment:
> I don't want to have to manually reset the global DataLoaders after each test.

I'm wondering how it would look to just do this reset vs. add this conceptual complexity? While cumbersome, it would be immediately clear when reviewing other tests what's going on with the singleton vs. having it buried here, especially when debugging.

Also, is this problem only when writing tests for the data loader itself, or writing tests for the various dispatched thunks?

Reply:
Yeah, the extra argument's type is part of the `ThunkAction` type that `ActionFunc` and friends are based off of. I intentionally didn't type it properly for now because I wanted to make sure that this approach doesn't cause any issues yet. It's part of the worrying that this is going to get used which has lead me to write multiple paragraphs explaining how this is used and why people should make extra sure they need it before using it 😅

I'd like to leave it as-is for now, but it'll be easy to fix once we've let it soak for a bit